app/default.go

```diff
@@ -51,3 +53,33 @@ func RegisterShutdownHandler(name string, fn ShutdownFunc, options ...interface{
 	}
 	defaultApp.RegisterShutdownHandler(name, fn, options...)
 }
+
+// IsReady returns if the default app is ready (to be used by kubernetes readyness probe)
+func IsReady() bool {
+	return defaultApp.Ready
+}
+
+// SetReady sets the app to ready state
+func SetReady() {
+	defaultApp.Ready = true
+}
+
+// SetUnready sets the app to unready state
+func SetUnready() {
```

Comment:
Wishlist for future versions, SetUnready and SetUnhealthy could receive a "reason" string that would be output by the endpoints.